### ISSUE TITLE
Project structuring

### ISSUE THREAD

**Issue by Trooksy** [nus-cs2030/2122-s1#321]
# Summary
What is the best way to tackle the project? Would it be possible to structure this project without using event classes?

## Description 
I have been trying to attempt the project and the way I structured it was without an event class. Just a customer class with a comparator for arrival times and a server class with a comparator for nextAvailableTime. the code works but different output. Now, the output is a bit different because server class sorts based on time and it should sort based on availability, else id. If I change my solution to sort based on that, the servers and customers become unlinked. Any tips?

## Screenshots (if any): 
Insert Images here if necessary

**Comment by ZChenxin** [comment]
I don’t think it’s possible to do the project without an event class, especially the later levels. The customer and server info can be stored in the event class so the unlinked problem does not occur

**Comment by chenghong99** [comment]
Yes  i do not think it is possible. I suggest u to make do with what you have now since you have less than 3 days to complete it

**Comment by yuehernkang** [comment]
Yes i think event is necessary. However I did the project without customer class

**Comment by albertsutz** [comment]
Hi, I think using Event Class is definitely useful. This is because you can use PriorityQueues to do the sorting for you (polling everytime). When you want to sort the event, you can create EventComparator class that can be put inside the priorityqueues, which will make your life easier basically :).

For Customer, I can see why you want to use Customer class (Maybe put which server it is now in). However, for me just using an integer (as the customer's id) is sufficient.

**Comment by audrey-eng** [comment]
I think an event class is helpful as well, since you can implement getServer and getCustomer methods in the event class to solve the unlinked problem you're facing

**Comment by Devvynneo** [comment]
I think you can create extra event classes! These could help you structure the system such as rest event!

**Comment by justinekoh** [comment]
It makes sense to include an Event class. Furthermore I read somewhere that design will be graded(?) so if thats the case having an Event class will definitely make your project design make more sense. Practically, event class can also be used to implement serverRests and serverBacks without changing too much of your design (if you do include Event class)

**Comment by DarrenCsAcc** [comment]
I dont think its possible it is structured to be done with an Event Class
